parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -147,9 +158,40 @@ public static Metrics footerMetrics(ParquetMetadata metadata, Stream<FieldMetric
       upperBounds.remove(fieldId);
     }
 
+    // populate upper and lower bounds for floating columns
+    updateFloatingColumnsBounds(fieldMetricsMap, metricsConfig, fileSchema, lowerBounds, upperBounds);
+
     return new Metrics(rowCount, columnSizes, valueCounts, nullValueCounts,
-        MetricsUtil.createNanValueCounts(fieldMetrics, metricsConfig, fileSchema),
-        toBufferMap(fileSchema, lowerBounds), toBufferMap(fileSchema, upperBounds));
+        MetricsUtil.createNanValueCounts(fieldMetricsMap.values().stream(), metricsConfig, fileSchema),
+        toBufferMap(fileSchema, lowerBounds),
+        toBufferMap(fileSchema, upperBounds));
+  }
+
+  private static void updateFloatingColumnsBounds(
+      Map<Integer, FieldMetrics> idToFieldMetricsMap, MetricsConfig metricsConfig, Schema schema,
+      Map<Integer, Literal<?>> lowerBounds, Map<Integer, Literal<?>> upperBounds) {
+    idToFieldMetricsMap.entrySet().forEach(entry -> {
+      int fieldId = entry.getKey();
+      FieldMetrics metrics = entry.getValue();
+      MetricsMode metricsMode = MetricsUtil.metricsMode(schema, metricsConfig, fieldId);
+
+      // only check for MetricsModes.None, since we don't truncate float/double values.
+      if (metricsMode != MetricsModes.None.get()) {
+        if (metrics.upperBound() == null) {
+          // upper and lower bounds will both null or neither
+          lowerBounds.remove(fieldId);
+          upperBounds.remove(fieldId);
+        } else if (metrics.upperBound() instanceof Float) {
+          lowerBounds.put(fieldId, Literal.of((Float) metrics.lowerBound()));
+          upperBounds.put(fieldId, Literal.of((Float) metrics.upperBound()));
+        } else if (metrics.upperBound() instanceof Double) {
+          lowerBounds.put(fieldId, Literal.of((Double) metrics.lowerBound()));
+          upperBounds.put(fieldId, Literal.of((Double) metrics.upperBound()));
+        } else {
+          throw new IllegalStateException("[BUG] Only Float/Double column metrics should be collected by Iceberg");
```

Comment:
Minor: I would probably rephrase this as "Expected only float or double column metrics" and throw an `UnsupportedOperationException`. I think that is (marginally) more clear because it isn't quite right to say that only float metrics should be collected. The code assumed that, but there isn't a reason for that to be the case other than we didn't implement the other types.